### PR TITLE
2.0rc27

### PR DESCRIPTION
**Pull request by hadim**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

ping @mdraw 

I have created this rc package so we don't have to wait for the official release of the 2.0. I am using rc versions from pip for some time now and it's very stable.

### CODE REVIEW

**Comment by hadim** [comment]
The CI error is because `requirements/base.txt` is not included in the pypi archive: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=147166&view=logs&j=1bf226d3-0e2f-52d8-fa93-7d9e633347b3&t=ec8d466d-e3b4-5115-4f06-222398f91e6c&l=449

```
Created temporary directory: /tmp/pip-install-wiid99i0
Processing $SRC_DIR
  Created temporary directory: /tmp/pip-req-build-_5yqcjhj
  Added file://$SRC_DIR to build tracker '/tmp/pip-req-tracker-udzl6nms'
    Created temporary directory: /tmp/pip-modern-metadata-tbewvxfa
    Running command /home/conda/feedstock_root/build_artifacts/omegaconf_1587474272804/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/bin/python /home/conda/feedstock_root/build_artifacts/omegaconf_1587474272804/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmp2gcva66u
    Preparing wheel metadata: started
    Traceback (most recent call last):
      File "/home/conda/feedstock_root/build_artifacts/omegaconf_1587474272804/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py", line 257, in <module>
        main()
      File "/home/conda/feedstock_root/build_artifacts/omegaconf_1587474272804/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py", line 240, in main
        json_out['return_val'] = hook(**hook_input['kwargs'])
      File "/home/conda/feedstock_root/build_artifacts/omegaconf_1587474272804/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py", line 110, in prepare_metadata_for_build_wheel
        return hook(metadata_directory, config_settings)
      File "/home/conda/feedstock_root/build_artifacts/omegaconf_1587474272804/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.6/site-packages/setuptools/build_meta.py", line 158, in prepare_metadata_for_build_wheel
        self.run_setup()
      File "/home/conda/feedstock_root/build_artifacts/omegaconf_1587474272804/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.6/site-packages/setuptools/build_meta.py", line 250, in run_setup
        self).run_setup(setup_script=setup_script)
      File "/home/conda/feedstock_root/build_artifacts/omegaconf_1587474272804/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.6/site-packages/setuptools/build_meta.py", line 143, in run_setup
        exec(compile(code, __file__, 'exec'), locals())
      File "setup.py", line 19, in <module>
        with pathlib.Path("requirements/base.txt").open() as requirements_txt:
      File "/home/conda/feedstock_root/build_artifacts/omegaconf_1587474272804/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.6/pathlib.py", line 1183, in open
        opener=self._opener)
      File "/home/conda/feedstock_root/build_artifacts/omegaconf_1587474272804/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.6/pathlib.py", line 1037, in _opener
        return self._accessor.open(self, flags, mode)
      File "/home/conda/feedstock_root/build_artifacts/omegaconf_1587474272804/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.6/pathlib.py", line 387, in wrapped
        return strfunc(str(pathobj), *args)
    FileNotFoundError: [Errno 2] No such file or directory: 'requirements/base.txt'

```

How do you guys want to fix that?

File: .ci_support/linux_.yaml
Comment (omry) [inline .ci_support/linux_.yaml:12]: OmegaConf 2.0 supports PYthon >= 3.6.
This does not seem to correctly represent that.

File: recipe/meta.yaml
Comment (omry) [inline recipe/meta.yaml]: I released rc25 today and will likely release rc26 tomorrow.

File: .ci_support/linux_.yaml
Comment (hadim) [reply .ci_support/linux_.yaml:12]: This file is autogenerated and should not be modified. I'll take care of the python requirements elsewhere.

File: recipe/meta.yaml
Comment (hadim) [reply recipe/meta.yaml]: I'll bump to the latest one before merging.

**Comment by hadim** [comment]
ready to be merged @mdraw @omry

File: recipe/meta.yaml
Comment (omry) [inline recipe/meta.yaml]: you can remove six.

Comment (omry) [reply recipe/meta.yaml]: https://github.com/omry/omegaconf/blob/master/requirements/base.txt

Comment (omry) [reply recipe/meta.yaml]: dataclass is needed for Python == 3.6

**Comment by omry** [comment]
now that the tests are included in the sdist, can you run them as a part of the package build process?

**Comment by omry** [comment]
Cool, thanks.

Edit the title (rc26)

**Comment by hadim** [comment]
For the tests please see: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=147926&view=logs&jobId=1bf226d3-0e2f-52d8-fa93-7d9e633347b3&j=1bf226d3-0e2f-52d8-fa93-7d9e633347b3&t=ec8d466d-e3b4-5115-4f06-222398f91e6c

Usually, tests are done when building the pypi package not the conda package. But it's ok if you like to have it here too.

**Comment by hadim** [comment]
> will this only be pulled in for Python 3.6?

No the package will be available for py >=3.6

**Comment by hadim** [comment]
As for the tests, any guidance to help fix them will be helpful (or we can disable them).

Another thing, I can also add you as a maintainer of the recipe if you wish.

**Comment by omry** [comment]
> > will this only be pulled in for Python 3.6?
> 
> No the package will be available for py >=3.6

the dependency on dataclasses is only in 3.6 (it's built into 3.7+).

**Comment by omry** [comment]
> As for the tests, any guidance to help fix them will be helpful (or we can disable them).
> 
> Another thing, I can also add you as a maintainer of the recipe if you wish.

taking a look.
and sure, please add me.
although I will likely not have the cycles to drive updates.

**Comment by hadim** [comment]
@conda-forge-admin, please rerender

**Comment by omry** [comment]
> As for the tests, any guidance to help fix them will be helpful (or we can disable them).
> 
> Another thing, I can also add you as a maintainer of the recipe if you wish.

I am not familiar with Azure, how do I see the test failures?

**Comment by omry** [comment]
There are two problems:
1. OmegaConf tests output warnings. I fixed it in master (I had an old warning filtering in my pytest.ini).
2. There appears to be some kind of system problem (see [this](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=147934&view=logs&j=f97ca392-d626-52d0-4b2e-4b27aa319a77&t=d276ba93-4510-59b4-3917-c75eb70f82df))

I think 1 is not supposed to fail the build or cause 2.

**Comment by hadim** [comment]
@conda-forge/python we have couple of issues here.

- Tests are failing only windows and it seems to be related to Windows path?
- This package is for python >=3.6 and I am not sure what is the best way to specify this in the recipe. Should that be `noarch`?

**Comment by hadim** [comment]
Ok, I think I got it this time.

@conda-forge/core can I get a review + merge, please? Not sure @mdraw is available.

**Comment by hadim** [comment]
@omry administrating this conda package is pretty simple. At each new pypi release a PR on this repo will be automatically created. Your job is simply to merge this PR if all CIs are green for each new pypi release.

**Comment by beckermr** [comment]
See this CFEP on rc releases: https://github.com/conda-forge/cfep/blob/master/cfep-05.md

**Comment by hadim** [comment]
This rc is very stable. Is that possible to merge it as a regular release?

**Comment by beckermr** [comment]
That is not our formal policy and IDK how to make exceptions. Many people do this. Can we ask upstream for a real release?

**Comment by hadim** [comment]
Ok it's fine I'll make a rc package.

**Comment by hadim** [comment]
It should be an RC package now. From my understanding, it must be merged in an `rc` branch and not `master`.

**Comment by omry** [comment]
OmegaConf 2.0 is breaking compatibility with OmegaConf 1.4 (latest stable).
The API is still subject to change between different release candidates without much of a notice.

**Comment by beckermr** [comment]
Then we should def use an rc label!

File: recipe/meta.yaml
Comment (omry) [inline recipe/meta.yaml]: is it possible to condition the dataclasses dependency on the python version being 3.6 ?

Comment (hadim) [reply recipe/meta.yaml]: I removed he python selector since it's the only way to get a noarch conda package for omegaconf. The dataclasses conda package is empty for python >3.6: https://github.com/conda-forge/dataclasses-feedstock/pull/3

Comment (hadim) [reply recipe/meta.yaml]: It's kind of hacky but I don't think we have another choice here. Maybe a conda-forge dev can comment on this?

Comment (omry) [reply recipe/meta.yaml]: This is okay for now but I would like to know that there is a plan to address this. the dataclasses package is bringing dataclasses to Python 3.6 and is not required in newer versions.

In pip land this is expressed as 
```
dataclasses;python_version=='3.6'
```

I would be good if Conda supported something similar.

Comment (hadim) [reply recipe/meta.yaml]: The alternative is to make this package not `noarch` but I don't know whether it's preferable or not on a conda-forge point of view. Ping @beckermr @CJ-Wright

Comment (beckermr) [reply recipe/meta.yaml]: noarch packages use less of our CI resources - that being said we are doing ok on them, so i'd say if you have a real need, then by all mean make the package non-noarch.

Comment (hadim) [reply recipe/meta.yaml]: @beckermr I have a lint error

`Non noarch packages should have python requirement without any version constraints.`

This package is also only for python >= 3.6.

Comment (omry) [reply recipe/meta.yaml]: I vote to make it an arch specific package then. what would that look like in the recipe?
can we special case Python 3.6 and have newer Python versions use the same definition?

**Comment by hadim** [comment]
There is also an error on Windows tests:

```
OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'D:\\bld\\omegaconf_1587660885620\\work' -> 'D:\\bld\\omegaconf_1587660885620\\work_moved_omegaconf-2.0.0rc26-py>=36_0_win-64_main_build_loop'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Miniconda\Scripts\conda-build-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Miniconda\lib\site-packages\conda_build\cli\main_build.py", line 474, in main
    execute(sys.argv[1:])
  File "C:\Miniconda\lib\site-packages\conda_build\cli\main_build.py", line 465, in execute
    verify=args.verify, variants=args.variants)
  File "C:\Miniconda\lib\site-packages\conda_build\api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 2929, in build_tree
    shutil.move(metadata.config.work_dir, dest)
  File "C:\Miniconda\lib\shutil.py", line 561, in move
    symlinks=True)
  File "C:\Miniconda\lib\shutil.py", line 321, in copytree
    os.makedirs(dst)
  File "C:\Miniconda\lib\os.py", line 220, in makedirs
    mkdir(name, mode)
OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: 'D:\\bld\\omegaconf_1587660885620\\work_moved_omegaconf-2.0.0rc26-py>=36_0_win-64_main_build_loop'
##[error]Cmd.exe exited with code '1'.
```

**Comment by omry** [comment]
@hadim, can you explain how this is addressing the dataclasses issue?

**Comment by hadim** [comment]
> @hadim, can you explain how this is addressing the dataclasses issue?

The selector `# [py36]` will only add `dataclasses` dep to python 3.6 installation not for the other python version. The issue is that those selectors `# [py36]` cannot be used when the package is `noarch`. At least this is my understanding. Maybe a core dev can correct me if I am wrong.

**Comment by omry** [comment]
> > @hadim, can you explain how this is addressing the dataclasses issue?
> 
> The selector `# [py36]` will only add `dataclasses` dep to python 3.6 installation not for the other python version. The issue is that those selectors `# [py36]` cannot be used when the package is `noarch`. At least this is my understanding. Maybe a core dev can correct me if I am wrong.

Gotcha.

By the way, looking at the yaml configs for the build, I can't help but think how much simpler the whole thing can be if it used OmegaConf for layering configs :)

**Comment by hadim** [comment]
This PR cannot be merged before the following are solved:

- [X] Windows CI tests issue: https://github.com/conda-forge/omegaconf-feedstock/pull/3#issuecomment-618526200
- [X] Python version constraint with non `noarch`: https://github.com/conda-forge/omegaconf-feedstock/pull/3#issuecomment-618513354

**Comment by hadim** [comment]
CIs are all green so I guess this PR is ok now.

Ready to be reviewed and merged.

**Comment by omry** [comment]
Great, can you bump to rc27?

**Review by mdraw — CHANGES_REQUESTED** [review]
The recipe itself looks good to me. You are still targeting the `master` branch though. Can you please open a new PR against the `rc` branch of this repo? See https://github.com/conda-forge/cfep/blob/master/cfep-05.md#specification

**Comment by hadim** [comment]
I can't target the `rc` branch since it does not exist. Can you make one?

**Comment by mdraw** [comment]
I just created a new `rc` branch here. Does it work now?

**Pull request closed without merging**
